pinecone_text/dense/openai_encoder.py

```diff
@@ -76,9 +80,13 @@ def _encode(
             )
 
         try:
-            response = self._client.embeddings.create(
-                input=texts_input, model=self._model_name
+            params: Dict[str, Any] = dict(
+                input=texts_input,
+                model=self._model_name,
             )
+            if self._dimension:
```

Comment:
either we should have here `self._dimension is not None` or better just assert on input that `dimension > 0`, right now if someone pass 0 it will work with OpenAI default which is kind of weird

Reply:
Right fixing